test/vault_helpers.go

```diff
@@ -487,6 +340,60 @@ func runVaultEC2AuthTest(t *testing.T, packerBuildName string) {
 	})
 }
 
+func setupAmi(t *testing.T, examplesDir string, packerBuildName string) {
+	awsRegion := aws.GetRandomRegion(t, nil, nil)
+	test_structure.SaveString(t, examplesDir, SAVED_AWS_REGION, awsRegion)
+
+	tlsCert := generateSelfSignedTlsCert(t)
+	saveTlsCert(t, examplesDir, tlsCert)
+
+	amiId := buildAmi(t, AMI_EXAMPLE_PATH, packerBuildName, tlsCert, awsRegion)
+	test_structure.SaveAmiId(t, examplesDir, amiId)
+}
+
+func teardownResources(t *testing.T, examplesDir string) {
+	terraformOptions := test_structure.LoadTerraformOptions(t, examplesDir)
+	terraform.Destroy(t, terraformOptions)
+
+	amiId := test_structure.LoadAmiId(t, examplesDir)
+	awsRegion := test_structure.LoadString(t, examplesDir, SAVED_AWS_REGION)
+	aws.DeleteAmi(t, awsRegion, amiId)
+
+	keyPair := test_structure.LoadEc2KeyPair(t, examplesDir)
+	aws.DeleteEC2KeyPair(t, keyPair)
+
+	tlsCert := loadTlsCert(t, examplesDir)
+	cleanupTlsCertFiles(tlsCert)
+}
+
+func deployCluster(t *testing.T, examplesDir string, uniqueId string, terraformVars map[string]interface{}) {
+	amiId := test_structure.LoadAmiId(t, examplesDir)
+	awsRegion := test_structure.LoadString(t, examplesDir, SAVED_AWS_REGION)
+
+	keyPair := aws.CreateAndImportEC2KeyPair(t, awsRegion, uniqueId)
+	test_structure.SaveEc2KeyPair(t, examplesDir, keyPair)
+
+	for key, value := range map[string]interface{}{
+		VAR_AMI_ID:                 amiId,
+		VAR_VAULT_CLUSTER_NAME:     fmt.Sprintf("vault-test-%s", uniqueId),
+		VAR_CONSUL_CLUSTER_NAME:    fmt.Sprintf("consul-test-%s", uniqueId),
+		VAR_CONSUL_CLUSTER_TAG_KEY: fmt.Sprintf("consul-test-%s", uniqueId),
+		VAR_SSH_KEY_NAME:           keyPair.Name,
+	} {
+		terraformVars[key] = value
+	}
+
+	terraformOptions := &terraform.Options{
+		TerraformDir: examplesDir,
+		Vars:         terraformVars,
+		EnvVars: map[string]string{
+			ENV_VAR_AWS_REGION: awsRegion,
+		},
```

Comment:
Given that we're dynamically generating all the vars, should we just pass in the awsRegion directly to the Terraform vars and avoid the use of EnvVars altogether?

Reply:
We could. I was just following the previous examples that use `data "aws_region" "current" {}` and get it through env. That could be removed and we can pass it as vars, but I honestly don't see what's the difference and what would be the advantage of doing that over what was being done before. Could you help me understand?

Reply:
> but I honestly don't see what's the difference

Actually, that was my thought exactly. But if there's no difference, and you see a different way of handling one particular var then it makes the reader question why it's being used that way (as I did). So I'm just trying to minimize the potential confusion readers will have.

Reply:
I see! I'll take a note to do this in a separate PR if you don't mind because I just did the vault auto-unseal branching off this one